### PR TITLE
Land ID generation and better Location format

### LINKED ISSUE #84: Implement Land ID Generation System
Currently, our land registry system uses block timestamps to generate land IDs, which can be problematic due to potential collisions and predictability. We need to implement a more robust and secure method for generating unique land IDs.

Task:
Develop and implement a new system for generating unique land IDs that doesn't rely on block timestamps. 

The new system should be:

•	Collision-resistant
•	Unpredictable
•	Scalable for a large number of land parcels
•	Suggested Approach:

Some options to consider:
•	Keccak256 hash of multiple inputs (e.g., owner address, location, current block number)
•	Incremental counter combined with a random element
•	UUID-like generation adapted for smart contract use
•	Implement the chosen method in the LandRegistry contract:
•	Create a new function for ID generation
•	Update the register_land function to use the new ID generation method
•	Ensure the new method is gas-efficient
•	Update any related functions or storage variables that depend on the land ID
•	Implement a migration strategy for existing land parcels (if applicable)
•	Update tests to reflect the new ID generation system

Definition of Done:

•	A new land ID generation system is implemented and integrated into the LandRegistry contract
•	The new system demonstrably produces unique, unpredictable IDs
•	All existing functionality using land IDs is updated and working correctly
•	Comprehensive tests are written to verify the new ID generation system
•	The generated IDs are not easily predictable based on public information
•	All existing tests pass with the new implementation
•	New tests specifically for ID generation are added and passing

### PR DESCRIPTION
**Pull request by SoarinSkySagar**
## Detailed Information

Checklist: 
 - [x] Implemented `create_land_id` function to randomize `land_id` generation (using Cairo inbuilt Poseidon hashing)
 - [x] Modified the snippets where the logic was dependent on the `land_id` being incremented by 1 when created
 - [x] Created a new test function to test the `land_id` generation
 - [x] Ensured that all tests are passing  

---

## Related Issues

Closes #84 

---

## Type of Change

- [x] 🐛 Bug fix or ⚙️ enhancement
- [ ] ✨ New feature or Chore (change with no new features or fixes)
- [ ] 📚 Documentation update

---

## Checklist (select as many as applicable)

- [x] The code follows the style guidelines of this project.
- [x] All new and existing tests pass.
- [x] This pull request is ready to be merged and reviewed.

### CODE REVIEW

**Comment by SoarinSkySagar** [comment]
Hi @fishonamos, please review!

**Comment by fishonamos** [comment]
Please update the pr title too.

**Comment by SoarinSkySagar** [comment]
@fishonamos okay, doing that. On an other note, I can't access the telegram group of landver, it says its a private group and now its not in my chats list as well.

File: land_registry/src/land_register.cairo [hunk 1]
```diff
@@ -211,5 +213,18 @@ pub mod LandRegistryContract {
...
+            let timestamp = get_block_timestamp();
+
+            let caller_hash = PoseidonTrait::new().update_with(caller).finalize();
+            let timestamp_hash = PoseidonTrait::new().update_with(timestamp).finalize();
+            let location_hash = PoseidonTrait::new().update_with(location).finalize();
+
+            let felt_land_id = caller_hash + timestamp_hash + location_hash;
+            let land_id: u256 = felt_land_id.into();
+            land_id
+        }
```
[the remark this answers is not in the document]
Comment (SoarinSkySagar) [reply land_registry/src/land_register.cairo:228]: > struct Location {
> latitude: f64,
> longitude: f64,
> }

But Cairo uses field elements so we can't store decimals/float values, what should I do in this case?

**Comment by SoarinSkySagar** [comment]
@fishonamos, since there aren't any float values in Cairo I have used felt252 values for latitude and longitude and refactored codebase wherever required to achieve the same functionality throughout the codebase.

**Comment by fishonamos** [comment]
> @fishonamos, since there aren't any float values in Cairo I have used felt252 values for latitude and longitude and refactored codebase wherever required to achieve the same functionality throughout the codebase.
Good. That fits the size for now. Can you move the create_id function outside the contract? Other wise. Looks good.

**Comment by fishonamos** [comment]
Good. That fits the size for now. Can you move the create_id function outside the contract? Other wise. Looks good.

**Pull request merged**